packages/dd-trace/src/appsec/iast/security-controls/parser.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const log = require('../../../log')
+const { getMarkFromVulnerabilityType, CUSTOM_SECURE_MARK } = require('../taint-tracking/secure-marks')
+
+const SECURITY_CONTROL_DELIMITER = ';'
+const SECURITY_CONTROL_FIELD_DELIMITER = ':'
+const SECURITY_CONTROL_ELEMENT_DELIMITER = ','
+
+const INPUT_VALIDATOR_TYPE = 'INPUT_VALIDATOR'
+const SANITIZER_TYPE = 'SANITIZER'
+
+const validTypes = [INPUT_VALIDATOR_TYPE, SANITIZER_TYPE]
+
+function parse (securityControlsConfiguration) {
+  const controls = new Map()
+
+  securityControlsConfiguration?.replace(/\s/g, '')
+    .split(SECURITY_CONTROL_DELIMITER)
+    .map(parseControl)
+    .filter(control => !!control)
+    .forEach(control => {
+      let controlsByFile = controls.get(control.file)
+      if (!controlsByFile) {
+        controlsByFile = []
+        controls.set(control.file, controlsByFile)
+      }
+      controlsByFile.push(control)
+    })
+
+  return controls
+}
+
+function parseControl (control) {
+  if (!control) return
+
+  const fields = control.split(SECURITY_CONTROL_FIELD_DELIMITER)
+
+  if (fields.length < 3 || fields.length > 5) {
+    // TODO: do we want telemetry log for these cases?
```

Comment:
I don't think so, but we could log an error rather than warn to be aware when the customer is using it incorrectly.